### PR TITLE
Update kind dependency to 0.5.1

### PR DESCRIPTION
**Pull request by sslavic**
See:
- https://github.com/kubernetes-sigs/kind/releases/tag/v0.5.1
- https://github.com/kubernetes-sigs/kind/releases/tag/v0.5.0

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: quay.io/giantswarm/golang:1.11.6
+      - image: quay.io/giantswarm/golang:1.12.9
```
Comment (rossf7) [inline .circleci/config.yml]: <3

**Review by rossf7 — APPROVED** [review]
LGTM once the build is sorted. Thx for getting this updated.

**Comment by MarcelMue** [comment]
Build is failing:
`build github.com/giantswarm/e2ectl: cannot load sigs.k8s.io/kind/pkg/cluster/config: cannot find module providing package sigs.k8s.io/kind/pkg/cluster/config`

**Comment by sslavic** [comment]
Works with both old retagged
```
e2ectl cluster create
time="2019-08-22T13:59:25+02:00" level=info msg="Pulling image: quay.io/giantswarm/kind-node:v1.14.3 ..."
 ✓ Ensuring node image (quay.io/giantswarm/kind-node:v1.14.3) 🖼 
 ✓ Preparing nodes 📦 
 ✓ Creating kubeadm config 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
 ✓ Waiting ≤ 2m0s for control-plane = Ready ⏳ 
 • Ready after 24s 💚
Cluster creation complete. You can now use the cluster with:

export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
kubectl cluster-info
```

and new upstream images:
```
e2ectl cluster create --image kindest/node --version v1.15.3
time="2019-08-22T14:03:14+02:00" level=info msg="Pulling image: kindest/node:v1.15.3 ..."
 ✓ Ensuring node image (kindest/node:v1.15.3) 🖼 
 ✓ Preparing nodes 📦 
 ✓ Creating kubeadm config 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
 ✓ Waiting ≤ 2m0s for control-plane = Ready ⏳ 
 • Ready after 24s 💚
Cluster creation complete. You can now use the cluster with:

export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
kubectl cluster-info
```

PR for retagging new upstream images follows.

**Pull request merged**